README.md

```diff
@@ -46,57 +61,114 @@ api("com.github.open-feature:kotlin-sdk:[ANY_BRANCH]-SNAPSHOT")
+Look [here](https://openfeature.dev/ecosystem?instant_search%5BrefinementList%5D%5Btype%5D%5B0%5D=Provider&instant_search%5BrefinementList%5D%5Btechnology%5D%5B0%5D=kotlin) for a complete list of available providers.
+If the provider you're looking for hasn't been created yet, see the [develop a provider](#develop-a-provider) section to learn how to build it yourself.
 
-## 🚀 Usage
+Once you've added a provider as a dependency, it can be registered with OpenFeature like this:
 
 ```kotlin
-    // configure a provider and get client
-    OpenFeatureAPI.setProvider(customProvider)
-    val client = OpenFeatureAPI.getClient()
+OpenFeatureAPI.setProvider(MyProvider())
+OpenFeatureAPI.getClient()
+```
 
-    // get a bool flag value
-    client.getBooleanValue("boolFlag", default = false)
-    
-    // get a bool flag value async
-    coroutineScope.launch {
-        WithContext(Dispatchers.IO) {
-            client.awaitProviderReady()
-        }
-        client.getBooleanValue("boolFlag", default = false)
-    }
+
+### Targeting
+
+Sometimes, the value of a flag must consider some dynamic criteria about the application or user, such as the user's location, IP, email address, or the server's location.
+In OpenFeature, we refer to this as [targeting](https://openfeature.dev/specification/glossary#targeting).
+If the flag management system you're using supports targeting, you can provide the input data using the [evaluation context](https://openfeature.dev/docs/reference/concepts/evaluation-context).
+
+```kotlin
+// set a value to the global context
+val evaluationContext = ImmutableContext(
+    targetingKey = session.getId,
+    attributes = mutableMapOf("region" to Value.String("us-east-1")))
+OpenFeatureAPI.setEvaluationContext(evaluationContext)
 ```
 
-### Events
+### Hooks
+
+[Hooks](https://openfeature.dev/docs/reference/concepts/hooks) allow for custom logic to be added at well-defined points of the flag evaluation life-cycle.
+Look [here](https://openfeature.dev/ecosystem/?instant_search%5BrefinementList%5D%5Btype%5D%5B0%5D=Hook&instant_search%5BrefinementList%5D%5Btechnology%5D%5B0%5D=kotlin) for a complete list of available hooks.
+If the hook you're looking for hasn't been created yet, see the [develop a hook](#develop-a-hook) section to learn how to build it yourself.
+
+<!-- TODO: code example of setting hooks at all levels -->
+
+### Eventing
 
 Events allow you to react to state changes in the provider or underlying flag management system, such as flag definition changes, provider readiness, or error conditions.
 Initialization events (`PROVIDER_READY` on success, `PROVIDER_ERROR` on failure) are dispatched for every provider.
 Some providers support additional events, such as `PROVIDER_CONFIGURATION_CHANGED`.
+
+⚠️ _The Eventing APIs are still a work in progress for this SDK, and subsjected to backwards-incompatible changes. Most of the remaining work revolves around consolidating event semantics that fit the mobile use-case._
```

Comment:
I am thinking it's good to leave some context on why this feature is marked as `in-progress`. Not sure this is the best way to achieve this though.

Reply:
I don't think we need to provide context on this one, since the users won't be able to do anything with it (contribute etc) or?

Reply:
I don't mind this being here but it's up to you. Perhaps you could an issue to track eventing API feedback and encourage people to leave feedback there.

Reply:
I removed this part. I will move this topic to Issues as a followup, as suggested 👍